clickable.py

```diff
@@ -1214,17 +1211,128 @@ def _build(self):
         self.run_container_command(gocommand)
 
 
-class CordovaClickable(Clickable):
+class CordovaClickable(CMakeClickable):
+    # Lots of this code was based off of this:
+    # https://github.com/apache/cordova-ubuntu/blob/28cd3c1b53c1558baed4c66cb2deba597f05b3c6/bin/templates/project/cordova/lib/build.js#L59-L131
+    def __init__(self, *args, **kwargs):
+        super(CMakeClickable, self).__init__(*args, **kwargs)
+
+        self.platform_dir = os.path.join(self.cwd, 'platforms/ubuntu/')
+
+        self._dirs = {
+                'build'  : '{}/{}/{}/build/' .format(self.platform_dir, self.config.sdk, self.build_arch),
+                'prefix' : '{}/{}/{}/prefix/'.format(self.platform_dir, self.config.sdk, self.build_arch),
+                'make'   : '{}/build'.format(self.platform_dir)
+        }
+
+        self.temp = self._dirs['build']
+
+        self.config.specificDependencies = True
+        self.config.dependencies = [ # Got this list from check_reqs or somewhere
+                "cmake",
+                "libicu-dev:armhf",
+                "pkg-config",
+                "qtbase5-dev:armhf",
+                "qtchooser",
+                "qtdeclarative5-dev:armhf",
+                "qtfeedback5-dev:armhf",
+                "qtlocation5-dev:armhf",
+                "qtmultimedia5-dev:armhf",
+                "qtpim5-dev:armhf",
+                "libqt5sensors5-dev:armhf",
+                "qtsystems5-dev:armhf"
+        ]
+
     def _build(self):
-        command = "cordova -d build ubuntu --device -- --framework={}".format(self.config.sdk)
-        subprocess.check_call(shlex.split(command), cwd=self.cwd)
+
+        if not os.path.isdir(self.platform_dir):
+            # fail when not using docker, need it anyways
+            assert not (self.config.container_mode or self.config.lxd or self.config.chroot), "Need docker"
+
+            cordova_docker_image = "beevelop/cordova:v7.0.0"
+            command = "cordova platform add ubuntu; chown -R {uid}:{uid} platforms plugins node_modules".format(uid=os.getuid())
+
+            # Can't use self.run_container_command because need root
```

Comment:
When cordova is run, it writes to `$HOME/.cordova`. If run as the local user id, this just expands out to `/.cordova`, which the current user does not have access to write to.

This could be solved by creating a docker image with that directory writable by everyone or something like that, but I opted for being able to run straight from the docker registry.

Reply:
You could also set the `$HOME` env var to something like `/tmp` when running the docker container. But that's not a big deal, it's more for my curiosity that I asked.